### PR TITLE
OCPBUGS-37668: [release-4.15] operator: LeaderElectionReleaseOnCancel

### PR DESCRIPTION
**Pull request by zeeke**
backport of:
- https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/556

This ensure the lease is released by the operator as soon as it gracefully stops.

### CODE REVIEW

File: deploy/operator.yaml [hunk 1]
```diff
@@ -41,6 +41,8 @@ spec:
           image: $SRIOV_NETWORK_OPERATOR_IMAGE
           command:
           - sriov-network-operator
+          args:
+          - --leader-elect=$OPERATOR_LEADER_ELECTION_ENABLE
```
Comment (SchSeba) [inline deploy/operator.yaml:45]: do we need this in the bundle also?

Comment (zeeke) [reply deploy/operator.yaml:45]: Openshift bundle already has `--leader-elect` option set by default

https://github.com/openshift/sriov-network-operator/blob/master/manifests/stable/sriov-network-operator.clusterserviceversion.yaml#L366

Comment (SchSeba) [reply deploy/operator.yaml:45]: great thanks!

btw I am almost sure I already asked that :P

**Comment by zeeke** [comment]
/hold

wait for the fix in:
- https://github.com/k8snetworkplumbingwg/sriov-network-operator/pull/760

**Pull request closed without merging**